src/dmd/mars.d

```diff
@@ -1579,7 +1579,9 @@ private bool parseCommandLine(const ref Strings arguments, const size_t argc, re
                 //      -color=on|off
                 if (p[6] == '=')
                 {
-                    if (strcmp(p + 7, "off") == 0)
+                    if (strcmp(p + 7, "auto") == 0)
+                        params.color = false;
```

Comment:
This assumes the default to be `true`. You should check `on` here instead and error out if not `auto` below.